com.unity.netcode.gameobjects/Runtime/Core/NetworkManager.cs

```diff
@@ -2123,6 +2151,35 @@ private void SyncTime()
 #endif
         }
 
+        internal struct DisconnectReasonMessage : INetworkMessage
+        {
+            public string Reason;
+
+            public void Serialize(FastBufferWriter writer)
+            {
+                writer.TryBeginWrite(sizeof(int) + FastBufferWriter.GetWriteSize(Reason));
```

Comment:
Should we put a check in here to see if TryBeginWrite succeeds?  You can grow this up to I think like writer.MaxCapacity (which I think that is around 1292 bytes) so you could check if the write size of Reason exceeds MaxCapacity and if so then throw an exception with the reason it is failing,
Currently, if a user exceeds the maximum capacity then it will just crash with no real (easy to understand) reason as to why it failed.

Reply:
I'm not willing to compromise proper disconnection logic on the server just because the client wouldn't get a valid reason. So, we can't throw. We could error, or we could warn. Least intrusive, to me, would be:

```
                if (writer.TryBeginWrite(sizeof(int) + FastBufferWriter.GetWriteSize(Reason)))
                {
                    writer.WriteValue(Reason.Length);
                    writer.WriteValue(Reason);
                }
                else
                {
                    writer.TryBeginWrite(sizeof(int) + FastBufferWriter.GetWriteSize(""));
                    writer.WriteValue(0);
                    writer.WriteValue("");
                    Debug.LogWarning("Disconnect reason didn't fit. Disconnected without sending a reason. Consider shortening the reason string.");
                }
```
what do you think about that ?

Reply:
Yeah.. that works perfect.  Much better than an exception for sure!